src/datachain/model/bbox.py

```diff
@@ -20,28 +30,258 @@ class BBox(DataModel):
+            - width and height define the size of the bounding box.
+
+        Normalized coordinates are floats between 0 and 1, representing the
+        relative position of the pixels in the image.
+
+        Returns:
+            list[float]: The bounding box coordinates in PASCAL VOC format
+                with normalized coordinates.
+        """
+        coords_normalized = normalize_coords(self.coords, img_size)
+        return [
+            coords_normalized[0],
+            coords_normalized[1],
+            coords_normalized[2] - coords_normalized[0],
+            coords_normalized[3] - coords_normalized[1],
+        ]
+
     @staticmethod
-    def from_dict(coords: dict[str, float], title: str = "") -> "BBox":
-        assert isinstance(coords, dict) and set(coords) == {
-            "x1",
-            "y1",
-            "x2",
-            "y2",
-        }, "Bounding box must be a dictionary with keys 'x1', 'y1', 'x2' and 'y2'."
-        return BBox.from_list(
-            [coords["x1"], coords["y1"], coords["x2"], coords["y2"]],
+    def from_yolo(
+        coords: Sequence[float],
+        title: str = "",
+        normalized_to: Optional[Sequence[int]] = None,
+    ) -> "BBox":
+        """
+        Create a bounding box from coordinates in YOLO format.
+
+        YOLO format represents bounding boxes as [x_center, y_center, width, height],
+        where:
+            - (x_center, y_center) are the coordinates of the box center.
+            - width and height define the size of the bounding box.
+
+        If the input coordinates are normalized (i.e., floats between 0 and 1),
+        they will be converted to absolute pixel values based on the provided
+        image size. The image size should be given as a tuple (width, height)
+        via the `normalized_to` argument.
+
+        Args:
+            coords (Sequence[float]): The bounding box coordinates.
+            title (str, optional): The title or label for the bounding box.
+                Defaults to "".
+            normalized_to (Sequence[int], optional): The reference image size
+                (width, height) for denormalizing the bounding box. If None (default),
+                the coordinates are assumed to be absolute pixel values.
+
+        Returns:
+            BBox: The bounding box object.
+        """
+        coords = (
+            validate_bbox(coords)
+            if normalized_to is None
+            else validate_bbox_normalized(coords, normalized_to)
+        )
+
+        x_center, y_center, width, height = coords
+        return BBox(
             title=title,
+            coords=[
+                round(x_center - width / 2),
+                round(y_center - height / 2),
+                round(x_center + width / 2),
+                round(y_center + height / 2),
+            ],
+        )
+
+    def to_yolo(self) -> list[int]:
+        """
+        Convert the bounding box to YOLO format.
```

Comment:
yolo is normalized by default, no? (at when you train something it asks it to be normalized) - let's refer to utils in albumentations lib to be sure about formats, naming, etc

Reply:
> yolo is normalized by default, no? 

Yes and no. It depends.

By default internally Yolo [stores result data](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/results.py#L1004) [not](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/results.py#L1062) [normalized](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/results.py#L1072) (in `[x1, y1, x2, y2]` format for bounding boxes, btw), also Yolo outputs is not normalized (see [here](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/results.py#L769)).

But model training requires bounding boxes to be in yolo format (`x_center, y_center, wigth, height]`, normalized).

So "yolo format is normalized" is for model training.